src/wat-writer.cc

```diff
@@ -56,6 +56,20 @@ static const uint8_t s_is_char_escaped[] = {
     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1};
 
+// This table matches the characters allowed by wast-lexer.cc for `symbol`.
+// The disallowed printable characters are: "(),;[]{} and <space>.
```

Comment:
@binji thanks a lot for investigating!  I was hoping a few additional characters would be allowed in the lexer, but oh well.

@alexcrichton and @shepmaster for your reference the above characters will be replaced by`_` in `wasm2wat`.  Out of these characters I know `(),[] and <space>` are used in rust unmangled symbol names.  Below are a few examples I was worried about but now that I see them it doesn't seem too bad.

```
<(i32, i32) as wasm::Wasm>::wasm::hfc09b9ea108eba04
<_i32__i32__as_wasm::Wasm>::wasm::hfc09b9ea108eba04

<&'a T as core::fmt::Debug>::fmt::h273375ad95a1eb7b
<&'a_T_as_core::fmt::Debug>::fmt::h273375ad95a1eb7b

<core::fmt::Write::write_fmt::Adapter<'a, T> as core::fmt::Write>::write_char::h25e7e7a8edf51d28
<core::fmt::Write::write_fmt::Adapter<'a__T>_as_core::fmt::Write>::write_char::h25e7e7a8edf51d28

<std::error::<impl core::convert::From<alloc::string::String> for alloc::boxed::Box<std::error::Error + core::marker::Sync + core::marker::Send + 'static>>::from::StringError as std::error::Error>::description::hdf1479d893d74cf2
<std::error::<impl_core::convert::From<alloc::string::String>_for_alloc::boxed::Box<std::error::Error_+_core::marker::Sync_+_core::marker::Send_+_'static>>::from::StringError as std::error::Error>::description::hdf1479d893d74cf2
```